engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java

```diff
@@ -164,7 +173,10 @@ private HttpClient createHttpClientWithSslContext(SslContextBuilder sslContextBu
     private RestTemplate createRestTemplateWithSslContext(SSLContextBuilder sslContextBuilder) throws NoSuchAlgorithmException, KeyManagementException {
         SSLContext sslContext = sslContextBuilder.build();
         SSLConnectionSocketFactory sslContextFactory = new SSLConnectionSocketFactory(sslContext);
-        CloseableHttpClient httpClient = HttpClients.custom().setSSLSocketFactory(sslContextFactory).build();
+
+        HttpClientBuilder httpClientBuilder = HttpClients.custom().setSSLSocketFactory(sslContextFactory);
+        if(hostNameVerificationDisabled) httpClientBuilder.setSSLHostnameVerifier(new NoopHostnameVerifier());
```

Comment:
That's great, but we also need to take a look at what Domenico provided - the security vulnerability

"If the Veracode finding actually makes sense, you guys could try customizing the Netty secure HTTP client creation similarly to how the reactor people do, and then agree with Ciro to mitigate the vulnerability as dealt with. E.g.:
    private HttpClient createHttpClientWithSslContext(SslContextBuilder sslContextBuilder) throws SSLException {
        SslContext sslContext = sslContextBuilder.build();
        return HttpClient.create().secure(p -> p.sslContext(sslContext).handlerConfigurator(handler -> {
            SSLEngine engine = handler.engine();
            SSLParameters sslParameters = engine.getSSLParameters();
            sslParameters.setEndpointIdentificationAlgorithm("HTTPS");
            engine.setSSLParameters(sslParameters);
        }));
    }"
    
Which means that we should add a hostname verification manually by default when we use a Netty httpClient directly

Reply:
updated